app/electron/runCmd.ts

```diff
@@ -237,3 +233,50 @@ export function setupRunCmdHandlers(mainWindow: BrowserWindow | null, ipcMain: E
   }
   ipcMain.on('run-command', (event, eventData) => handleRunCommand(event, eventData, mainWindow));
 }
+
+/**
+ * Like CommandData, but everything is optional because it's not validated yet.
+ */
+type CommandDataPartial = Partial<CommandData>;
+
+/**
+ * Checks to see if it's what we expect.
+ */
+export function validateCommandData(eventData: CommandDataPartial): [boolean, string] {
+  if (!eventData || typeof eventData !== 'object' || eventData === null) {
+    return [false, `Invalid eventData data received: ${eventData}`];
+  }
+  if (typeof eventData.command !== 'string' || !eventData.command) {
+    return [false, `Invalid eventData.command: ${eventData.command}`];
+  }
+  if (!Array.isArray(eventData.args)) {
+    return [false, `Invalid eventData.args: ${eventData.args}`];
+  }
+  if (typeof eventData.options !== 'object' || eventData.options === null) {
+    return [false, `Invalid eventData.options: ${eventData.options}`];
+  }
+  if (typeof eventData.permissionSecrets !== 'object' || eventData.permissionSecrets === null) {
+    return [
+      false,
+      `Invalid permission secrets, it is not an object: ${typeof eventData.permissionSecrets}`,
+    ];
+  }
+  for (const [key, value] of Object.entries(eventData.permissionSecrets)) {
+    if (typeof value !== 'number') {
+      return [false, `Invalid permission secret for ${key}: ${typeof value}`];
+    }
+  }
+
+  const validCommands = ['minikube', 'az', 'scriptjs'];
```

Comment:
If we validate commands, but inside scriptjs we can run other commands, will this check prevent anything? (or maybe I am missing some details of how the scripts are run)

Reply:
Validation here is still useful in case one of the other parts does not work to prevent them. Inputs should be validated anyway.